### DISCUSSION TITLE
Should Open CSP 2.0 support SMW with Elastic Search 8.x ?

### DISCUSSION THREAD

**Discussion by Designburo** [Open-CSP/open-csp#5] [Polls]
Elastic Search is as of 7.x officially no longer Open Source. The discussion about switching to Open Search within the MediaWiki community is still on-going.
All current supported Elastic Search versions are end-of-life. Adding support for Elastic Search 8.x to Semantic MediaWiki would solve this problem.
There are obstacles as e.g. Cirrus Search does not support higher Elastic Search versions.
Taking it back to Open CSP ( That does not use Cirrus Search, but you could have installed it ), I'd like to know if we should add the support for ES 8.
Please also comment with why it should or should not.

**Comment by AdSvS** [comment]
It seems that we are using Elasicsearch in a way that complies with their SSPL license. Perhaps we should check this.
This means that Elasticsearch to us is as open source as it used to be. Although some consider it ‘fauxpen’ as it is not certified by the OSI.
If our use complies with their SSPL I am in favour of adding support for ES 8 and not move to Open Search.

Ad

**Comment by krabina** [comment]
While I think it is a good idea, in general I believe we should/have to follow whatever direction MediaWiki/CirrusSearch goes and that seems to be going to OpenSearch.

**Comment by AdSvS** [comment]
If MediaWiki/CirrusSearch is going for Open Search then that is important information but I thought that wasn’t clear.
It seems that Open Search can also be used as a vector database, so both products have what it takes to take us to the future.
